cloudinit/config/schema.py

```diff
@@ -301,8 +320,10 @@ def _schemapath_for_cloudconfig(config, original_content):
     return schema_line_numbers
 
 
-def _get_property_type(property_dict):
-    """Return a string representing a property type from a given jsonschema."""
+def _get_property_type(property_dict: dict) -> Union[str, bytes]:
```

Comment:
If SCHEMA_UNDEFINED can be of type `str` rather than type `bytes`, this annotation would be simpler and the changes below would not be necessary. On line 307 if `property_dict.get('enum')` returns a falsy value, then SCHEMA_UNDEFINED is returned from this function unmodified as type `bytes`. Alternatively casting `property_type` to `str` would also accomplish basically the same thing.